src/librustc_unicode/char.rs

```diff
@@ -110,53 +110,121 @@ impl Iterator for CaseMappingIter {
 #[stable(feature = "rust1", since = "1.0.0")]
 #[lang = "char"]
 impl char {
-    /// Checks if a `char` parses as a numeric digit in the given radix.
+    /// Checks if a `char` is a digit in the given radix.
+    ///
+    /// A 'radix' here is sometimes also called a 'base'. A radix of two
+    /// indicates a binary number, a radix of ten, decimal, and a radix of
+    /// sixteen, hexicdecimal, to give some common values. Arbitrary
+    /// radicum are supported.
     ///
     /// Compared to `is_numeric()`, this function only recognizes the characters
     /// `0-9`, `a-z` and `A-Z`.
     ///
-    /// # Return value
+    /// 'Digit' is defined to be only the following characters:
     ///
-    /// Returns `true` if `c` is a valid digit under `radix`, and `false`
-    /// otherwise.
+    /// * `0-9`
+    /// * `a-z`
+    /// * `A-Z`
+    ///
+    /// For a more comprehensive understanding of 'digit', see [`is_numeric()`][is_numeric].
+    ///
+    /// [is_numeric]: #method.is_numeric
     ///
     /// # Panics
     ///
-    /// Panics if given a radix > 36.
+    /// Panics if given a radix larger than 36.
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
-    /// let c = '1';
+    /// let d = '1';
     ///
-    /// assert!(c.is_digit(10));
+    /// assert!(d.is_digit(10));
     ///
-    /// assert!('f'.is_digit(16));
+    /// let d = 'f';
+    ///
+    /// assert!(d.is_digit(16));
+    /// assert!(!d.is_digit(10));
+    /// ```
+    ///
+    /// Passing a large radix, causing a panic:
+    ///
+    /// ```
+    /// use std::thread;
+    ///
+    /// let result = thread::spawn(|| {
+    ///   let d = '1';
+    ///
+    ///   d.is_digit(37);
```

Comment:
The indentation here looks like 2 spaces instead of 4, also perhaps this could have something like `// this panics` and the test could be annotated with `should_panic`?